azure/application-gateway/variables.tf

```diff
@@ -0,0 +1,207 @@
+variable "sku" {
+  description = "Properties relating to the SKU of the Applicaton Gateway"
+  type = object({
+    name     = string
+    tier     = string
+    capacity = optional(string)
+  })
+  default = {
+    name = "Standard_v2"
+    tier = "Standard_v2"
+  }
+}
+
+variable "autoscale" {
+  description = "Properties relating to the Autoscalling of the Applicaton Gateway"
+  type = object({
+    min = string
+    max = string
+  })
+  default = {
+    max = "3"
+    min = "1"
+  }
+}
+
+variable "resource_group_name" {
+  description = "The resource group name"
+  type        = string
+}
+
+variable "location" {
+  description = "The location where the Application Gateway will run "
+  type        = string
+}
+
+variable "application_gateway_name" {
+  description = "Name of the Application Gateway"
+  type        = string
+}
```

Comment:
Replace with `name` - we should know that this module creates an application gateway, so just calling it `name` should be pretty implicit.